pkg/ebpf/printk_patcher_test.go

```diff
@@ -107,9 +107,18 @@ func TestPatchPrintkNewline(t *testing.T) {
 	foundLines := []string{}
 
 	for _, line := range expectedLines {
-		actualLine, err := traceReader.ReadString('\n')
-		foundLines = append(foundLines, actualLine)
-		require.NoError(t, err)
+		// We allow up to two lines that don't match, to avoid failures with other tests outputting to
+		// the trace pipe.
+		var actualLine string
+		for allowedMisses := 0; allowedMisses < 2; allowedMisses++ {
```

Comment:
Instead of this can we pattern match and only process the lines generated by our test programs?

Reply:
I thought about it, but we'd have the same problem of how many lines we allow to not match our pattern before failing. Also, the compiler will output different assembly depending on the string itself, so I'd prefer to keep them as they are to ensure we're testing all the possible cases.